src/SparseVariationalApproximationModule.jl

```diff
@@ -386,4 +401,159 @@ function _prior_kl(sva::SparseVariationalApproximation{NonCentered})
     return (trace_term + m_ε'm_ε - length(m_ε) - logdet(C_ε)) / 2
 end
 
+# Pseudo-Observation Parametrisations of q(u).
+
+@doc raw"""
+    PseudoObsSparseVariationalApproximation(
+        likelihood, f::AbstractGP, z::AbstractVector
+    )
+
+Parametrises `q(f(z))`, the approximate posterior at `f(z)`, using a surrogate likelihood,
+`likelihood`: `q(f(z)) ∝ p(f(z)) likelihood(f(z))`.
+"""
+struct PseudoObsSparseVariationalApproximation{
+    Tlikelihood,Tf<:AbstractGP,Tz<:AbstractVector
+} <: AbstractSparseVariationalApproximation
+    likelihood::Tlikelihood
+    f::Tf
+    z::Tz
+end
+
+_get_prior(approx::PseudoObsSparseVariationalApproximation) = approx.f
+
+@doc raw"""
+    ObsCovLikelihood(S::AbstractMatrix{<:Real}, y::AbstractVector{<:Real})
+
+Chooses `likelihood(u) = N(y; u, S)`. `length(y)` must be equal to the number of
+pseudo-points utilised in the sparse variational approximation.
+"""
+struct ObsCovLikelihood{TS<:AbstractMatrix{<:Real},Ty<:AbstractVector{<:Real}}
+    S::TS
+    y::Ty
+end
+
+@doc raw"""
+    PseudoObsSparseVariationalApproximation(
+        f::AbstractGP,
+        z::AbstractVector,
+        S::AbstractMatrix{<:Real},
+        y::AbstractVector{<:Real},
+    )
+
+Convenience constuctor.
+Equivalent to
+```julia
+PseudoObsSparseVariationalApproximation(ObsCovLikelihood(S, y), f, z)
+```
+"""
+function PseudoObsSparseVariationalApproximation(
+    f::AbstractGP, z::AbstractVector, S::AbstractMatrix{<:Real}, y::AbstractVector{<:Real}
+)
+    return PseudoObsSparseVariationalApproximation(ObsCovLikelihood(S, y), f, z)
+end
+
+function AbstractGPs.posterior(
+    approx::PseudoObsSparseVariationalApproximation{<:ObsCovLikelihood}
+)
+    f = approx.f
+    z = approx.z
+    y = approx.likelihood.y
+    S = approx.likelihood.S
+    return posterior(f(z, S), y)
+end
+
+function _prior_kl(approx::PseudoObsSparseVariationalApproximation{<:ObsCovLikelihood})
+    f = approx.f
+    z = approx.z
+    y = approx.likelihood.y
+    S = approx.likelihood.S
+
+    # log marginal probability of pseudo-observations.
+    logp_pseudo_obs = logpdf(f(z, S), y)
+
+    # pseudo-reconstruction term.
+    m, C = mean_and_cov(posterior(approx)(z))
+    S_chol = cholesky(AbstractGPs._symmetric(S))
+    quad_form = sum(abs2, S_chol.U' \ (y - m))
+    pseudo_lik = -(length(y) * AbstractGPs.log2π + logdet(S_chol) + quad_form) / 2
+    trace_term = tr(S_chol \ C) / 2
+    return -logp_pseudo_obs + pseudo_lik - trace_term
+end
+
+@doc raw"""
+    DecoupledObsCovLikelihood(
+        S::AbstractMatrix{<:Real}, v::AbstractVector, y::AbstractVector{<:Real}
+    )
+
+Chooses `likelihood(u) = N(y; f(v), S)` where `length(y)` need not be equal to the number
```

Comment:
I don't understand how this makes any sense. On the left-hand side you have `u`, on the right hand side you don't. How are the two coupled ?

Reply:
Ah, so it's implicit via `f`. `u := f(z)`, so by making noisy observations of `f(v)`, you learn something about `f(z)`.

Reply:
....so then the inducing points are actually `v`? why are `u`/`z` needed then?